zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/LearnerHandler.java

```diff
@@ -968,6 +984,7 @@ protected long queueCommittedProposals(Iterator<Proposal> itr,
     public void shutdown() {
         // Send the packet of death
         try {
+            queuedPackets.clear();
```

Comment:
This seems unrelated to this patch, but it looks the right thing to do. Just want to make sure this was not accidentally introduced.

Reply:
Good catch! It is actually added in this patch to handle diff sync throttling. Diff txns are queued in syncFollower() before the throttle. Later when we decide that this diff sync should be throttled, the txns are in the queues already so we need to remove them.